mitiq/qse/tests/test_qse.py

```diff
@@ -32,14 +31,161 @@ def execute_with_depolarized_noise(circuit: QPROGRAM) -> np.ndarray:
     )
 
 
+def batched_execute_with_depolarized_noise(circuits) -> List[np.ndarray]:
+    return [execute_with_depolarized_noise(circuit) for circuit in circuits]
+
+
 def execute_no_noise(circuit: QPROGRAM) -> np.ndarray:
     return compute_density_matrix(
         convert_to_mitiq(circuit)[0], noise_level=(0,)
     )
 
 
+def test_execute_with_qse_no_noise():
+    qc = prepare_logical_0_state_for_5_1_3_code()
+    (
+        check_operators,
+        code_hamiltonian,
+    ) = get_5_1_3_code_check_operators_and_code_hamiltonian()
+    observable = get_observable_in_code_space(PauliString("ZZZZZ"))
+    mitigated_value = execute_with_qse(
+        qc, execute_no_noise, check_operators, code_hamiltonian, observable
+    )
+    assert np.isclose(mitigated_value.real, 1)
+
+    observable = get_observable_in_code_space(PauliString("XXXXX"))
+    mitigated_value = execute_with_qse(
+        qc, execute_no_noise, check_operators, code_hamiltonian, observable
+    )
+    assert np.isclose(mitigated_value.real, 0.5)
+
+
+def test_execute_with_qse():
+    qc = prepare_logical_0_state_for_5_1_3_code()
+    (
+        check_operators,
+        code_hamiltonian,
+    ) = get_5_1_3_code_check_operators_and_code_hamiltonian()
+    observable = get_observable_in_code_space(PauliString("ZZZZZ"))
+    mitigated_value = execute_with_qse(
+        qc,
+        execute_with_depolarized_noise,
+        check_operators,
+        code_hamiltonian,
+        observable,
+    )
+    assert abs(mitigated_value.real - 1) < 0.001
```

Comment:
What about if we also compare the QSE error to the unmitigated error?
e.g.

`assert abs(mitigated_value.real - 1) < abs(unmitigated_value.real - 1) `

In this way we could test if there is a positive improvement.